drivers/storage/portworx/deployment.go

```diff
@@ -796,16 +796,24 @@ func (t *template) getEnvList() []v1.EnvVar {
 	}
 
 	if t.cluster.Spec.ImagePullSecret != nil && *t.cluster.Spec.ImagePullSecret != "" {
-		envMap["REGISTRY_CONFIG"] = &v1.EnvVar{
-			Name: "REGISTRY_CONFIG",
-			ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
-					Key: ".dockerconfigjson",
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: *t.cluster.Spec.ImagePullSecret,
+		pxVer2_3_2, _ := version.NewVersion("2.3.2")
+		if t.pxVersion.LessThan(pxVer2_3_2) {
+			envMap["REGISTRY_CONFIG"] = &v1.EnvVar{
+				Name: "REGISTRY_CONFIG",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						Key: ".dockerconfigjson",
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: *t.cluster.Spec.ImagePullSecret,
+						},
```

Comment:
Do we actually need this block  (ie. is the current version of Px-Operator supposed to handle the px-2.3.2, released Dec 18, 2019)?

Reply:
There could be customers who are using 2.3.2. Don't want to break this for them. Will keep it around for a while.

Reply:
Oh.. didn't realize the latest PX-Operator has to support _all_ Portworx versions:
* code might get busy dealing with the backward-compatibility tweaks, and the new features
* alternatively, we could just introduce support-matrix for PX-Operator, so a certain PX-Operator would support certain set of PX-versions
   - in fact, I think it's very likely we'll end up with such version-specific deployments in the future

Reply:
Sure we can have that in future when it gets unmanageable.